feat: add support for consent

HejdaJakub opened the pull request:
* New object ConsentDefinition was created and added to parseRpcMethod.
* Five new tables were created to support consent.
* New object ConsentDefinition was added to openapi.yml.

BREAKING CHANGE: update DB version

Johaney-s commented on perun-db/postgres.sql:
```diff
@@ -1550,6 +1550,58 @@ create table groups_to_register (
+create table consents_definitions (
+	id integer not null,
+	name varchar not null,
+	decription varchar,
```
```suggestion
	description varchar,
```

Johaney-s commented on perun-db/postgres.sql:
```diff
@@ -1550,6 +1550,58 @@ create table groups_to_register (
+	modified_by varchar default user not null,
+	created_by_uid integer,
+	modified_by_uid integer,
+	constraint con_def_pk primary key(id)
```
I think we should also add constraint for the name to be unique, what do you think?

Johaney-s commented on perun-db/postgres.sql:
```diff
@@ -1550,6 +1550,58 @@ create table groups_to_register (
+	consent_def_id integer not null,   --identifier of consent definition
+	constraint service_consent_pk primary key (service_id,consent_def_id),
+  	constraint service_consent_service_fk foreignkey(service_id) references services(id),
+  	constraint service_consent_cons_def_fk foreign ke y (consent_def_id) references consents_definitions(id)
```
```suggestion
  	constraint service_consent_cons_def_fk foreign key (consent_def_id) references consents_definitions(id)
```

Johaney-s commented on perun-db/postgres.sql:
```diff
@@ -1550,6 +1550,58 @@ create table groups_to_register (
+	service_id integer not null,       --identifier of service
+	consent_def_id integer not null,   --identifier of consent definition
+	constraint service_consent_pk primary key (service_id,consent_def_id),
+  	constraint service_consent_service_fk foreignkey(service_id) references services(id),
```
```suggestion
  	constraint service_consent_service_fk foreign key(service_id) references services(id),
```

Johaney-s commented on perun-openapi/openapi.yml:
```diff
@@ -1385,6 +1385,19 @@ components:
+    ConsentDefinition:
+      properties:
+        name: { type: string }
+        desciption: { type: string }
```
```suggestion
        description: { type: string }
```

Johaney-s reviewed (changes requested):
Looks good, just a few suggestions. Also, I think the database changes should be also reflected in `test-schema.sql` and `postgresChangelog.txt`.

cuadradek commented on perun-db/postgres.sql:
```diff
@@ -1550,6 +1550,58 @@ create table groups_to_register (
+);
+
+-- FACILITY_CONSENT - relation between facilities and consent definitions
+create table facility_consent (
```
I think this would be a little more accurate, what do you think? Same for other tables.
```suggestion
create table facilities_consent_definitions (
```

HejdaJakub replied:
You're right, I edited the names of these tables.

cuadradek reviewed (commented):
I think we could create DB indices on each foreign key column which should speedup joining tables in queries. Also, I am not sure, but maybe we could add auditing columns (created_at, created_by, ...) to all tables. What do you think, @HejdaJakub @Johaney-s ?

HejdaJakub commented:
Latest changes:
 - added auditing columns to `posgres.sql`,` test-schema.sql`, and `postgresChangelog.txt`
 - added indexes on all foreign keys
 - hash code changed
 - changed the `description` to `text` and made it required

cuadradek commented on perun-base/src/test/resources/test-schema.sql:
```diff
@@ -1552,6 +1552,84 @@ create table groups_to_register (
+);
+
+-- CONSENTS - relation between users and consent definitions
+create table consents (
```
One last thing - we might want to have a column for ID in this table too. I am not sure if it is necessary, but I assume the Consent object will extend the PerunBean object, which has an ID attribute. What do you think?

cuadradek commented on perun-base/src/test/resources/test-schema.sql:
```diff
@@ -1552,6 +1552,84 @@ create table groups_to_register (
+	user_id integer not null,		   --identifier of user
+	consent_def_id integer not null,   --identifier of consent definition
+	status varchar not null,
+	createdDate timestamp not null,
```
Now that we have auditing columns, this column seems redundant at this moment.

HejdaJakub commented:
The architecture of consents has been changed, so the new class and basic objects are implemented in https://github.com/CESNET/perun/pull/3517.
